### PR TITLE
Documentation for path conflict validation

### PR DESCRIPTION
**Pull request by barchw**
**Description**

Changes proposed in this pull request:

- Add documentation for path conflict validation

**Pre-Merge Checklist**

- [ ] As a PR reviewer, verify code coverage and evaluate if it is acceptable.

**Related issues**

### CODE REVIEW

File: docs/contributor/02-00-path-validation.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+# Path conflict validation in APIRule v2alpha1
```
Comment (nataliasitko) [inline docs/contributor/02-00-path-validation.md:1]:
```suggestion
# Path Conflict Validation in APIRule `v2alpha1`
```

File: docs/contributor/02-00-path-validation.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+In the second case, the algorithm does not need to check the current segment, as the `{*}` operator matches any segment.
+The search continues recursively with the next segment in the path.
+
+If the third case occurs, and the current segment is a `{**}` operator, the algorithm checks if the path ends with any of the suffixes stored in the current node.
+The suffix search is performed from the segment that has the same index as the current segment in the checked exact path.
+If the suffix is found, the algorithm concludes that the paths conflict.
+
+If the search did not finish in any of the cases above, the algorithm continues until there are no more segments in the checked path.
+Then, the algorithm checks if the current node is a possible end of the path. (In the previously presented graph, those nodes are marked with a double circle.)
```
Comment (nataliasitko) [inline docs/contributor/02-00-path-validation.md:69]:
```suggestion
Then, the algorithm checks if the current node is a possible end of the path. In the previously presented graph, those nodes are marked with a double circle.
```

**Review by nataliasitko — APPROVED** [review]

File: docs/contributor/02-00-path-validation.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+However, all the children of the current node are considered as possible paths that can be taken.
+The algorithm recursively checks all the children of the current node in a DFS fashion,
+and if any of them contain a valid path, the algorithm concludes that the paths conflict.
+
+### Paths that Contains the `{**}` Operator
+
+In case the path that is currently being inserted contains the `{**}` operator,
+the algorithm is interrupted after the `{**}` operator is found (as it must be the last operator in the path).
+Then, the algorithm checks if there are any paths that end in the same suffix as the currently checked path.
+Example: If the path that is currently being inserted is `b/{**}/a/b`, the algorithm checks if any paths after the `b` node end in `/a/b`.
```
Comment (mluk-sap) [inline docs/contributor/02-00-path-validation.md:83]: > the algorithm checks if any paths after the `b` node end in `/a/b`.

should be:
the algorithm checks if any path after the `b` node ends in `/a/b`.

File: docs/contributor/02-00-path-validation.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+2. The current search node is a `{*}` node.
+3. The current search node is a `{**}` node.
+
+In the first case, the algorithm checks if the current segment is equal to the currently checked segment in the path.
+If so, the search continues recursively with the next segment in the path.
+
+In the second case, the algorithm does not need to check the current segment, as the `{*}` operator matches any segment.
+The search continues recursively with the next segment in the path.
+
+If the third case occurs, and the current segment is a `{**}` operator, the algorithm checks if the path ends with any of the suffixes stored in the current node.
```
Comment (mluk-sap) [inline docs/contributor/02-00-path-validation.md:63]: It would be nice to give an example here, like in the last section. WDYT?

**Review by mluk-sap — APPROVED** [review]

**Review by nataliasitko — APPROVED** [review]

**Pull request merged**